ports/stm32/mprprocport.c

```diff
@@ -0,0 +1,253 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2023 Arduino SA
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * modremoteproc stm32 port.
+ */
+#include <stdio.h>
+#include <stdint.h>
+
+#include "py/obj.h"
+#include "py/runtime.h"
+#include "py/gc.h"
+#include "py/stream.h"
+#include "extmod/vfs.h"
+
+#include "metal/alloc.h"
+#include "metal/errno.h"
+#include "metal/io.h"
+#include "metal/sys.h"
+#include "metal/device.h"
+#include "metal/utilities.h"
+#include "openamp/remoteproc.h"
+#include "openamp/remoteproc_loader.h"
+
+#define debug_printf(...)   // mp_printf(&mp_plat_print, __VA_ARGS__)
+
+#define FILE_STORE_BUF_SIZE   (1024)
+
+typedef struct filestore {
+    size_t len;
+    uint8_t *buf;
+    mp_obj_t file;
+} filestore_t;
+
+static int store_open(void *store, const char *path, const void **image_data) {
```

Comment:
It seems like this function, and the one below (and others in this file maybe) are generic enough that they could be put in `extmod/modrproc.c`?

Reply:
They are now indeed, thanks to switching to the VFS functions. I can move them to `modrproc.c` as a generic file store enabled/disabled via config (ex `MICROPY_PY_OPENAMP_FILE_STORE`) so ports can still override them (if needed) by providing their own `image_store_ops`.

Reply:
Done. I think it would be nice if the file store is moved to a separate file, ex `modrproc_store.c`, where maybe other stores can be added.